### ISSUE TITLE
[Feature]: tensor parallelism support for bnb quantization (via IBM's fork)

### ISSUE THREAD

**Issue by BlairSadewitz** [dphnAI/sonar#767]
### 🚀 The feature, motivation and pitch

I don't know if it's feasible or worthwhile to merge [this](https://github.com/IBM/vllm/tree/9855b99502c7537db5ef018129e603650800ac46), as maybe the trees are too divergent, etc., but cherry-picking commits for projects I don't fully understand is somehow a pastime for me, so ...

### Alternatives

I could always use one of the other 8.4234234*10^23 quantization methods, but, hey, variety is the spice of life--or something.

### Additional context

It doesn't work for pre-quantized models.  🎉~

**Comment by AlpinDale** [comment]
Perhaps, I'll have to look into it. bnb hasn't been a priority

**Comment by BlairSadewitz** [comment]
Yeah, I hear you. I'm gonna file a better PR in a second, though, so ... ;-)

**Comment by AlpinDale** [comment]
FYI I'm working on new kernels for massively speeding up bnb quants + add TP support for them. You might want to hold on for now, or help out with that upcoming PR if you're comfortable with CUDA

**Comment by AlpinDale** [comment]
Closing as part of the maintainer-approved issue cleanup. BitsAndBytes in-flight quantization supports tensor parallelism. Prequantized checkpoints remain restricted, as noted in the original request.
